cpp/include/cudf/wrappers/timestamps.hpp

```diff
@@ -45,6 +45,7 @@ struct timestamp : time_point<Duration> {
    *
    * @param other The `timestamp` to copy
    */
+  // TODO: This explict truncation is intended?
```

Comment:
would this one *and* the constructor `constexpr timestamp(Duration d)` have to be explicitly defined? they are:

- either relaxing or adding explicit specifiers that aren't in the chrono's `time_point` type
- allows narrowing/truncation of types, when it is disallowed by `std::chrono`

is it not possible to inherit them from the base type?

Reply:
Yeah the list of timestamp constructors evolved over time, and in response to various compiler errors (@karthikeyann reminds me of one I forgot in [this comment](https://github.com/rapidsai/cudf/pull/5359#discussion_r435774513)). Some of them may not be valid reasons any longer.

I'd say change/remove whichever ones we want, and recompile to see what breaks (this can be frustrating, as changing this header triggers a near full re-compile).